packages/state-transition/src/block/processBlsToExecutionChange.ts

```diff
@@ -10,11 +10,16 @@ export function processBlsToExecutionChange(
   state: CachedBeaconStateCapella,
   signedBlsToExecutionChange: capella.SignedBLSToExecutionChange
 ): void {
-  assertValidBlsToExecutionChange(state, signedBlsToExecutionChange, true);
-
   const addressChange = signedBlsToExecutionChange.message;
-  const validator = state.validators.get(addressChange.validatorIndex);
 
+  const validation = isValidBlsToExecutionChange(state, signedBlsToExecutionChange, true);
+  if (!validation.valid) {
+    throw (
+      validation.error ?? Error(`Invalid signedBlsToExecutionChange validatorIndex=${addressChange.validatorIndex}`)
```

Comment:
Type the return as `{valid: true} | {valid: false, error: Error}` so you don't need to default on a random error here.